create-doi-request/src/main/java/no/unit/nva/doirequest/create/CreateDoiRequestHandler.java

```diff
@@ -0,0 +1,80 @@
+package no.unit.nva.doirequest.create;
+
+import static nva.commons.core.attempt.Try.attempt;
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClientBuilder;
+import com.amazonaws.services.lambda.runtime.Context;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.time.Clock;
+import java.util.Map;
+import no.unit.nva.identifiers.SortableIdentifier;
+import no.unit.nva.publication.service.impl.DoiRequestService;
+import no.unit.nva.publication.service.impl.UserInstance;
+import no.unit.nva.publication.service.impl.exceptions.BadRequestException;
+import no.unit.nva.publication.service.impl.exceptions.InternalServerErrorException;
+import nva.commons.apigateway.ApiGatewayHandler;
+import nva.commons.apigateway.RequestInfo;
+import nva.commons.apigateway.exceptions.ApiGatewayException;
+import nva.commons.apigateway.exceptions.ConflictException;
+import nva.commons.core.Environment;
+import nva.commons.core.attempt.Failure;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CreateDoiRequestHandler extends ApiGatewayHandler<CreateDoiRequest, Void> {
+
+    public static final String DOI_ALREADY_EXISTS_ERROR = "A Doi request already exists";
+    private static final Logger LOGGER = LoggerFactory.getLogger(CreateDoiRequestHandler.class);
+    private final DoiRequestService doiRequestService;
+
+    public CreateDoiRequestHandler() {
+        this(new DoiRequestService(AmazonDynamoDBClientBuilder.defaultClient(),
+                Clock.systemDefaultZone()),
+            new Environment());
+    }
+
+    public CreateDoiRequestHandler(DoiRequestService requestService, Environment environment) {
+        super(CreateDoiRequest.class, environment, LOGGER);
+        this.doiRequestService = requestService;
+    }
+
+    @Override
+    protected Void processInput(CreateDoiRequest input, RequestInfo requestInfo, Context context)
+        throws ApiGatewayException {
+        URI customerId = requestInfo.getCustomerId().map(URI::create).orElse(null);
```

Comment:
I have a feeling that these are in some ways default for API-GW methods and should be provided by the library…probably wrong about this, but I mention it anyway to see if it triggers something useful in your mind

Reply:
The only thing we can do is the conversion of the String to a URI in the commons. But what happens when the value is absent is maybe the responsibility of the caller.  One maybe wishes to throw an exception, another may wish to continue doing something else.